experimental/framework/tree-react-api/src/useTree.ts

```diff
@@ -41,3 +45,40 @@ export function useTree<TRoot extends FieldSchema>(
 
 	return typedTree.context.root as unknown as SchemaAware.TypedField<TRoot>;
 }
+
+/**
+ * React Hook to trigger invalidation of the current component if anything in the document changes.
+ */
+export function useTreeContext(document: TreeContext): void {
+	// This proof-of-concept implementation allocates a state variable this is modified
+	// when the tree changes to trigger re-render.
+	const [invalidations, setInvalidations] = React.useState(0);
+
+	// Register for tree deltas when the component mounts
+	React.useEffect(() => {
+		// Returns the cleanup function to be invoked when the component unmounts.
+		return document.on("afterChange", () => {
+			setInvalidations(invalidations + 1);
+		});
+	}, [document]);
```

Comment:
I think this should have `invalidations` as a dep.

Reply:
You can also do this:
```suggestion
	React.useEffect(() => {
		// Returns the cleanup function to be invoked when the component unmounts.
		return document.on("afterChange", () => {
			setInvalidations((invalidations) => invalidations + 1);
		});
	}, [document]);
```

Reply:
The first option means it will needlessly allocated a new on change callback after every change which is needless overhead, but the second option (which I did not know was possible) works perfectly!